libnmstate/nm/connection.py

```diff
@@ -21,85 +21,120 @@
 from . import nmclient
 
 
-def create_profile(settings):
-    con_profile = nmclient.NM.SimpleConnection.new()
-    for setting in settings:
-        con_profile.add_setting(setting)
-
-    return con_profile
+class ConnectionProfile(object):
+
+    def __init__(self, profile=None):
+        self._con_profile = profile
+        self._nmclient = nmclient.client()
+        self._mainloop = nmclient.mainloop()
+
+    def create(self, settings):
```

Comment:
`setting` input should be changed in following PR/s to a base Setting object.
(As the change is incremental, only after all setting type will use it, we can change it here)